Sources/VLCVideoOptionsControlBar.swift

```diff
@@ -0,0 +1,142 @@
+
+@objc protocol VLCVideoOptionsControlBarDelegate: class {
+    func didToggleFullScreen(_ optionsBar: VLCVideoOptionsControlBar)
+    func didToggleRepeat(_ optionsBar: VLCVideoOptionsControlBar)
+    func didSelectSubtitle(_ optionsBar: VLCVideoOptionsControlBar)
+    func didSelectMoreOptions(_ optionsBar: VLCVideoOptionsControlBar)
+    func didToggleInterfaceLock(_ optionsBar: VLCVideoOptionsControlBar)
+}
+
+class VLCVideoOptionsControlBar: UIStackView {
+    
+    // MARK: Instance variables
+    @objc var delegate: VLCVideoOptionsControlBarDelegate?
+    private var __repeatMode: VLCRepeatMode = .doNotRepeat
+    
+    @objc var orientationAxis: NSLayoutConstraint.Axis {
+        set {
+            // rotate the control bar's orientation by switching it's height and width values and changing it's layout positioning
+            axis = newValue
+            let minX: CGFloat = frame.maxX - frame.size.height
+            let minY: CGFloat = frame.maxY - frame.size.width
+            frame = CGRect(x: minX, y: minY, width: frame.size.height, height:frame.size.width)
+        }
+        
+        get {
+            return axis
+        }
+    }
+    
+    @objc var repeatMode: VLCRepeatMode {
+        set {
+            __repeatMode = newValue
+            switch newValue {
+            case .repeatCurrentItem:
+                repeatButton.setImage(UIImage(named: "repeatOne-new"), for: .normal)
+                break
+            case .repeatAllItems:
+                repeatButton.setImage(UIImage(named: "repeat-new"), for: .normal)
+                break
+            default: // no repeat
+                repeatButton.setImage(UIImage(named: "no-repeat-new"), for: .normal)
+                break
+            }
+        }
+        
+        get {
+            return __repeatMode
+        }
+    }
+    
+    @objc var toggleFullScreenButton: UIButton = {
+        var toggle = UIButton(type: .system)
+        toggle.addTarget(self, action: #selector(toggleFullscreen), for: .touchUpInside)
+        toggle.setImage(UIImage(named: "fullscreenIcon-new"), for: .normal)
+        toggle.tintColor = .white
+        //TODO: add accessability options for fullScreenButton
+        return toggle
+    }()
+    
+    @objc var selectSubtitleButton: UIButton = {
+        var subbutton = UIButton(type: .system)
+        subbutton.addTarget(self, action: #selector(selectSubtitle), for: .touchUpInside)
+        subbutton.setImage(UIImage(named: "subtitleIcon-new"), for: .normal)
+        subbutton.tintColor = .white
+        // TODO: add accessability options for selectingSubtitleButton
+        return subbutton
+    }()
+    
+    @objc var repeatButton: UIButton = {
+        var rptButton = UIButton(type: .system)
+        rptButton.addTarget(self, action: #selector(toggleRepeat), for: .touchUpInside)
+        rptButton.setImage(UIImage(named: "no-repeat-new"), for: .normal)
+        rptButton.tintColor = .orange
```

Comment:
Is there a specific reason of why this is set to orange?

Reply:
Nope, my mistake